### ISSUE TITLE
Explict release configuration should disable inferring release targets (multiple-tokens error)

### ISSUE THREAD

**Issue by dekimsey** [goreleaser/goreleaser#3361]
### Is your feature request related to a problem? Please describe.

I use my personal access token to release code to some projects on GitLab (private) and GitHub (public). Currently, when goreleaser has access to both tokens via `~/.config/goreleaser/gitlab_token` and `~/.config/goreleaser/github_token` respectively the application fails to deploy with an error message:

> multiple tokens found, but only one is allowed: GITHUB_TOKEN, GITLAB_TOKEN
> 
> Learn more at https://goreleaser.com/errors/multiple-tokens

### Describe the solution you'd like

My project's goreleaser.yml has an explicit `release` block configured. I'd like it's presence to indicate that _this scm_ should be used, and that the inferring logic be disabled.

### Describe alternatives you've considered

I tried setting the env_files.gitlab_token to a custom path but that also does not work. It seems the presence of both tokens makes it impossible to opt-out of the default targets.

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
seems valid, yes, thanks for the suggestion!

**Comment by caarlos0** [comment]
its a bit weird, but you could, in theory, set the path to a file that does not exist for the token you don't want to use, e.g.:

```yaml
# .goreleaser.yaml
env_files:
  gitlab_token: /nope
```

seems like an usable workaround....

**Comment by caarlos0** [comment]
I think that this is actually a pretty good fix, anything else I can think of is pretty much the same config with different semantics...

I'll just update the docs explaining this.

**Comment by dekimsey** [comment]
Is null a valid value here?

**Comment by caarlos0** [comment]
it will be interpreted as a path to a file named `null`, but other than that, should be fine, yes

**Comment by dekimsey** [comment]
Hrmm, I think that'd be odd behavior as `null` is keyword in yaml.

**Comment by caarlos0** [comment]
> Hrmm, I think that'd be odd behavior as `null` is keyword in yaml.

ahh right, then it'll probably default to empty, which goreleaser will default the default file path for each token.

my suggestion would be to set to something like `~/.this-file-does-not-exist` or something clear enough for future you as well as for other people...

**Comment by caarlos0** [comment]
if all token provider mechanisms fails, it will fail as well... it ignores file not found errors because of the defaults, and that error per se is not that relevant, whereas if that error happens and its also not possible to load any others, it means things will fail later, so it errors.

anyway, for now, will proceed as said before as its 100% backward-compatible with no tricks needed.

thanks!

**Comment by dekimsey** [comment]
Makes sense, thinking about this morning I thought similarly. It's documented now, that'll work just fine.

Thank you!
